infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -104,13 +114,47 @@
   }
 
   /**
-   * Creates specified pod.
+   * Starts the specified Pod via a Deployment.
    *
-   * @param pod pod to create
+   * @param pod pod to deploy
    * @return created pod
    * @throws InfrastructureException when any exception occurs
    */
   public Pod create(Pod pod) throws InfrastructureException {
+    putLabel(pod, CHE_WORKSPACE_ID_LABEL, workspaceId);
+    ObjectMeta metadata = pod.getMetadata();
+    PodSpec podSpec = pod.getSpec();
+    podSpec.setRestartPolicy("Always"); // Only allowable value
+    try {
+      Deployment deployment =
+          clientFactory
+              .create(workspaceId)
+              .extensions()
+              .deployments()
+              .createNew()
+              .withMetadata(metadata)
+              .withNewSpec()
+              .withReplicas(1)
+              .withNewTemplate()
+              .withMetadata(metadata)
+              .withSpec(podSpec)
+              .endTemplate()
+              .endSpec()
+              .done();
+      return awaitAndGetPod(deployment);
+    } catch (KubernetesClientException e) {
+      throw new KubernetesInfrastructureException(e);
+    }
+  }
+
+  /**
+   * Create a terminating pod that is not part of a Deployment.
```

Comment:
The class is called Deployment but methods create a pod that is not a part of the deployment. I think it may be a bit confusing.

Reply:
I agree, see the comment above about renaming to `create[Not]Terminating()` to better reflect the intended purpose of this class.